src/common/CommandLine.cs

```diff
@@ -0,0 +1,1820 @@
+                    ThrowIfNotFirst("NoSpaceOnQualifierValues");
+                _noSpaceOnQualifierValues = value;
+            }
+        }
+        /// <summary>
+        /// If the positional parameters might look like named parameters (typically happens when the tail of the
+        /// command line is literal text), it is useful to stop the search for named parameters at the first
+        /// positional parameter. 
+        /// 
+        /// Because some parameters sets might want this behavior and some might not, you specify the list of
+        /// parameter sets that you want to opt in.
+        /// 
+        /// The expectation is you do something like
+        ///     * commandLine.ParameterSetsWhereQualifiersMustBeFirst = new string[] { "parameterSet1" };
+        ///     
+        /// The empty string is a wildcard that indicats all parameter sets have the qualifiersMustBeFirst 
+        /// attribute.   This is the only way to get this attribute on the default parameter set.  
+        /// </summary>
+        public string[] ParameterSetsWhereQualifiersMustBeFirst
+        {
+            get { return _parameterSetsWhereQualifiersMustBeFirst; }
+            set
+            {
+                ThrowIfNotFirst("ParameterSetsWhereQualifiersMustBeFirst");
+                NoSpaceOnQualifierValues = true;
+                _parameterSetsWhereQualifiersMustBeFirst = value;
+            }
+        }
+        /// <summary>
+        /// By default qualifiers may being with a - or a / character.   Setting code:QualifiersUseOnlyDash will
+        /// make / invalid qualifier marker (only - can be used)
+        /// </summary>
+        public bool QualifiersUseOnlyDash
+        {
+            get { return _qualifiersUseOnlyDash; }
+            set
+            {
+                if (_qualifiersUseOnlyDash != value)
+                    ThrowIfNotFirst("OnlyDashForQualifiers");
+                _qualifiersUseOnlyDash = value;
+            }
+        }
+
+
+        // TODO remove?   Not clear it is useful.  Can be useful for CMD.EXE alias (which provide a default) but later user may override.  
+        /// <summary>
+        /// By default, a non-list qualifier can not be specified more than once (since one or the other will
+        /// have to be ignored).  Normally an error is thrown.  Setting code:LastQualifierWins makes it legal, and
+        /// the last qualifier is the one that is used.  
+        /// </summary>
+        public bool LastQualifierWins
+        {
+            get { return _lastQualifierWins; }
+            set { _lastQualifierWins = value; }
+        }
+
+        // These routines are typically are not needed because ParseArgsForConsoleApp does the work.  
+#if !COREFX
+        public CommandLineParser() : this(Environment.CommandLine) { }
+#endif
+        public CommandLineParser(string commandLine)
+        {
+            ParseWords(commandLine);
+        }
+        public CommandLineParser(string[] args)
```

Comment:
Is there a reason we cannot switch to using the arg overload here instead of doing the forking in GenFacades?